BinaryEncoding.md

```diff
@@ -65,20 +65,40 @@ Yes:
 
 ## Global structure
 
-* A module contains:
-  * a header followed by
-  * a table (sorted by offset) containing, for each section, its type and offset (within the module), followed by
-  * a sequence of sections.
+* A module contains (in this order):
+  - A header, containing:
+    + The [magic number](https://en.wikipedia.org/wiki/Magic_number_%28programming%29)
+    + Other data TBD (possibly entrypoint, memory bitness, source language, etc.)
+  - A table (sorted by offset) containing, for each section:
+    + A string literal section type name
+    + 32-bit offset within the module
```

Comment:
64 bit is fine with me. that's why I didn't specify originally, I thought this might be decided based on experimental data. do we expect modules that are larger than 4GB? That feels like a pretty insane amount of code.